### PR TITLE
Migrate macOS test app to react-native-test-app

### PR DESCRIPTION
**Pull request by tido64**
### Platforms Impacted
- [ ] iOS
- [x] macOS
- [ ] win32
- [ ] windows
- [ ] android

### Description of changes

Migrates macOS test app to `react-native-test-app`.

### Verification

| Before                                       | After                                      |
|----------------------------------------------|--------------------------------------------|
| image | Screenshot 2020-07-14 at 10 09 49 |

### Pull request checklist

This PR has considered (when applicable):
- [x] Automated Tests
- [ ] Documentation and examples
- [ ] Keyboard Accessibility
- [ ] Voiceover
- [ ] Internationalization and Right-to-left Layouts

### CODE REVIEW

**Comment by tido64** [comment]
cc @HeyImChris, @Saadnajmi

File: apps/macos/app.json [hunk 1]
```diff
@@ -1,4 +1,7 @@
 {
   "name": "FluentTester-macos",
-  "displayName": "FluentTester-macos"
+  "displayName": "FluentTester-macos",
+  "components": [
```
Comment (HeyImChris) [inline apps/macos/app.json:4]: What is this doing?

File: apps/macos/package.json [hunk 2]
```diff
@@ -19,8 +19,9 @@
     "@fluentui-react-native/tester": "^0.0.1",
     "@uifabricshared/theming-react-native": "0.7.67",
     "@uifabricshared/themed-stylesheet": "0.3.67",
-    "react": "16.11.0",
-    "react-native-macos": "^0.61.0-0"
+    "react-native": "0.61.5",
```
Comment (HeyImChris) [inline apps/macos/package.json:22]: Shouldn't need react-native if we're pulling in react-native-macos?

File: apps/macos/app.json (hunk 1 above)
Comment (tido64) [reply apps/macos/app.json:4]: `components`? It's to declare the apps that you registered with `AppRegistry`. The list is used to populate the React menu so you can switch between them. There's only one app in the FluentUI test app so it's not very useful here. You can see it in use here:

react-native-test-app-macos

File: apps/macos/react-native.config.js [hunk 3]
```diff
@@ -18,5 +18,10 @@
 const path = require('path');
 
 module.exports = {
-  reactNativePath: path.dirname(require.resolve('react-native-macos/package.json')),
+  project: {
+    ios: {
+      project: 'src/FluentUITester.xcworkspace'
+    }
+  },
```
Comment (Saadnajmi) [inline apps/macos/react-native.config.js:25]: Why is this iOS over here?

Comment (tido64) [reply apps/macos/react-native.config.js:25]: It's a bug in `@react-native-community/cli`. I've opened an issue in https://github.com/react-native-community/cli/issues/1054.

Comment (Saadnajmi) [reply apps/macos/react-native.config.js:25]: So we still need this for the macOS test app for autolinking, even though we don't have an iOS project here?

Comment (tido64) [reply apps/macos/react-native.config.js:25]: Yes, I haven't looked much into it but I believe macOS autolinking reuses much of the iOS code.

**Pull request merged**